arroyo-sql/src/expressions.rs

```diff
@@ -166,287 +170,318 @@ impl Expression {
+                    )),
+                    BuiltinScalarFunction::NullIf => {
+                        Ok(Expression::DataStructure(DataStructureFunction::NullIf {
+                            left: Box::new(arg_expressions.remove(0)),
+                            right: Box::new(arg_expressions.remove(0)),
+                        }))
+                    }
+                    BuiltinScalarFunction::MakeArray => {
+                        if matches!(arg_expressions[0].return_type(), TypeDef::StructDef(_, _)) {
+                            bail!("make_array only supports primitive types");
+                        };
+                        Ok(Expression::DataStructure(DataStructureFunction::MakeArray(
+                            arg_expressions,
+                        )))
+                    }
+                    BuiltinScalarFunction::Struct | BuiltinScalarFunction::ArrowTypeof => {
+                        bail!("data structure function {:?} not implemented", fun)
+                    }
+                    BuiltinScalarFunction::DatePart
+                    | BuiltinScalarFunction::DateTrunc
+                    | BuiltinScalarFunction::ToTimestamp
+                    | BuiltinScalarFunction::ToTimestampMillis
+                    | BuiltinScalarFunction::ToTimestampMicros
+                    | BuiltinScalarFunction::ToTimestampSeconds
+                    | BuiltinScalarFunction::DateBin
+                    | BuiltinScalarFunction::CurrentDate
+                    | BuiltinScalarFunction::FromUnixtime
+                    | BuiltinScalarFunction::Now
+                    | BuiltinScalarFunction::CurrentTime => {
+                        bail!("date function {:?} not implemented", fun)
+                    }
+                    BuiltinScalarFunction::Digest
+                    | BuiltinScalarFunction::MD5
+                    | BuiltinScalarFunction::Random
+                    | BuiltinScalarFunction::SHA224
+                    | BuiltinScalarFunction::SHA256
+                    | BuiltinScalarFunction::SHA384
+                    | BuiltinScalarFunction::SHA512 => Ok(HashExpression::new(
+                        fun.clone(),
+                        Box::new(arg_expressions.remove(0)),
+                    )?),
+                    BuiltinScalarFunction::ToHex => bail!("hex not implemented"),
+                    BuiltinScalarFunction::Uuid => bail!("UUID unimplemented"),
+                    BuiltinScalarFunction::Cbrt => bail!("cube root unimplemented"),
+                    BuiltinScalarFunction::Degrees => bail!("degrees not implemented yet"),
+                    BuiltinScalarFunction::Pi => bail!("pi not implemented yet"),
+                    BuiltinScalarFunction::Radians => bail!("radians not implemented yet"),
+                }
+            }
+            Expr::ScalarUDF { fun, args } => match fun.name.as_str() {
+                "get_first_json_object" => {
```

Comment:
Right now I can register a function that overwrites builtins, any easy way to return a better error? More specifically, right now my UDF gets added to the ArroyoSchemaProvider, planned against, and then returns an error because it hits this block.